portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -94,7 +90,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         String comment = entity.getProperty("comment").toString();
         String useremail = entity.getProperty("userEmail").toString();
         long timestamp = Long.parseLong(entity.getProperty("timestamp").toString());
-        messages.add(new Comment(comment, useremail, timestamp));
+        try {
+            // If there is a nickname
+            String nickname = entity.getProperty("nickname").toString();
+            messages.add(new Comment(comment, useremail, nickname, timestamp));
+        } catch (NullPointerException e) {
```

Comment:
Exceptions should be for exceptional circumstances. Not having a nickname set seems like a basic normal usecase, so it would be much better style to explicitly check if the nickname is set, rather than catching the null pointer exception.

Also, NullPointerException is a RuntimeException, so you generally don't want to catch it. If this exception manifests, it's usually because there's a but in the code.